test/service/test_response.py

```diff
@@ -3873,3 +3818,539 @@ def test_file_urls(self):
+                    'laboratory': 'Department of Nephrology',
+                    'correspondingContributor': True,
+                    'email': '1376785378@qq.com'
+                },
+                {
+                    'institution': 'Xiangya Hospital, Central South University',
+                    'contactName': 'Yong,,Zhong',
+                    'projectRole': None,
+                    'laboratory': 'Department of Nephrology',
+                    'correspondingContributor': True,
+                    'email': 'zhongyong121@163.com'
+                },
+                {
+                    'institution': 'EMBL-EBI',
+                    'contactName': 'Ida,,Zucchi',
+                    'projectRole': 'data curator',
+                    'laboratory': None,
+                    'correspondingContributor': None,
+                    'email': 'idazucchi@ebi.ac.uk'
+                }
+            ],
+            'publications': [
+                {
+                    'publicationTitle': 'Intrarenal Single-Cell Sequencing of'
+                                        ' Hepatitis B Virus Associated'
+                                        ' Membranous Nephropathy.',
+                    'officialHcaPublication': False,
+                    'publicationUrl': 'https://doi.org/10.3389/fmed.2022.869284',
+                    'doi': '10.3389/fmed.2022.869284'
+                }
+            ],
+            'supplementaryLinks': [None],
+            'matrices': {},
+            'contributedAnalyses': {},
+            'accessions': [
+                {'namespace': 'insdc_project', 'accession': 'SRP366901'},
+                {'namespace': 'geo_series', 'accession': 'GSE199850'},
+                {'namespace': 'insdc_study', 'accession': 'PRJNA821712'}
+            ],
+            'accessible': True
+        }
+        self.assertEqual(expected_project, project)
+
+    @cached_property
+    def project_field_names(self) -> list[FieldName]:
+        return [
+            k
+            for k, v in self.app_module.app.metadata_plugin.field_mapping.items()
+            if v[:2] == ('contents', 'projects')
+        ]
+
+    def get_project_field_value(self, hit: JSON, field: str) -> Any:
+        project = one(hit['projects'])
+        if field in ['contactName', 'institution', 'laboratory']:
+            if project['contributors']:
+                return [contributor[field] for contributor in project['contributors']]
+            else:
+                return None
+        elif field == 'publicationTitle':
+            if project['publications']:
+                return [publication[field] for publication in project['publications']]
+            else:
+                return None
+        elif field == 'project':
+            return project['projectShortname']
+        elif field == 'projectEstimatedCellCount':
+            return project['estimatedCellCount']
+        else:
+            return project[field]
```

Comment:
This is essentially the inverse of `HCASearchResponseStage.make_projects`. Putting that here undoes the effort of having an abstraction in the first place. If we merged this, we'd be taking on technical debt.